backend/app/routes/search.py

```diff
@@ -5,33 +5,39 @@
 from fastapi import HTTPException, Query
 from httpx import AsyncClient
 
-from app.config import SOLR_URL
+from app.solr.operations import search
 
 from .util import DEFAULT_SORT, internal_api_router
 
 
+# pylint: disable=too-many-arguments
 @internal_api_router.get("/search", name="apis:get-search")
 async def search_get(
-    q: str = Query(..., description="Query string"),
     collection: str = Query(..., description="Collection"),
-    rows: int = Query(10, description="Row count", gte=0, le=100),
-    cursor: str = Query("*", description="Cursor"),
+    q: str = Query(..., description="Query string"),
+    qf: list[str] = Query([], description="Query fields", example=["authors", "title"]),
+    fq: list[str] = Query(
+        [],
+        description="Filter query",
+        example=["journal:Geonomos", 'journal:"Solar Energy"'],
+    ),
     sort: list[str] = Query(
         [], description="Sort order", example=["description asc", "name desc"]
     ),
+    rows: int = Query(10, description="Row count", gte=0, le=100),
+    cursor: str = Query("*", description="Cursor"),
 ):
     """Do a search against the specified collection"""
```

Comment:
Would be better to rename the function to `serach_in_collection` or `get_from_collection`  instead of adding a comment? The path will be the same.

Reply:
Here I went for matching the HTTP path section and verb. Docstring can then provide extra info.